unit-tests/live/d400/test-hdr.py

```diff
@@ -0,0 +1,638 @@
+
+# CHECKING HDR AFTER PIPE RESTART
+
+with test.closure("HDR Running - restart hdr at restream", "[hdr][live][using_pipeline]"):
+    ctx = rs.context()
+    device = test.find_first_device_or_exit()
+    if ctx:
+        devices_list = ctx.query_devices()
+        device_count = devices_list.size()
+        if devices_list.size() != 0:
+            depth_sensor = device.first_depth_sensor()
+            if depth_sensor and depth_sensor.supports(rs.option.hdr_enabled):
+
+                cfg = rs.config()
+                cfg.enable_stream(rs.stream.depth)
+                pipe = rs.pipeline()
+                pipe.start(cfg)
+
+                depth_sensor.set_option(rs.option.hdr_enabled, 1)
+                test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+
+                for i in range(10):
+                    data = pipe.wait_for_frames()
+
+                test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+                pipe.stop()
+
+                pipe.start(cfg)
+                test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+                pipe.stop()
+
+                depth_sensor.set_option(rs.option.hdr_enabled, 0)  # disable hdr before next tests
+        else:
+            log.e("No device was found - skipping test")
+
+
+# helper method
+# checks that the frames resulting from the hdr_merge processing block are generated
+# from frames streamed within this method (and not using old frames)
+# returned value: False if some metadata could not be read - otherwise, True
+def check_hdr_frame_counter(pipe, num_of_frames, merging_filter):
+    min_counter = -1
+    min_counter_set = False
+    for i in range(num_of_frames):
+        data = pipe.wait_for_frames()
+
+        # get depth frame data
+        depth_frame = data.get_depth_frame()
+        if not depth_frame.supports_frame_metadata(rs.frame_metadata_value.sequence_id):
+            log.e("Frame metadata not supported")
+            return False
+
+        depth_counter = depth_frame.get_frame_metadata(rs.frame_metadata_value.frame_counter)
+
+        if depth_counter < min_counter:  # in case of frame counter reset during stream this might happen
+            log.e("Frame counter reset!")
+            min_counter_set = False
+
+        if not min_counter_set:
+            min_counter = int(depth_counter)
+            min_counter_set = True
+
+        # apply HDR Merge process
+        merged_frame = merging_filter.process(data)
+
+        # get hdr frame data
+        if not merged_frame.supports_frame_metadata(rs.frame_metadata_value.sequence_id):
+            log.e("Frame metadata not supported")
+            return False
+
+        hdr_counter = merged_frame.get_frame_metadata(rs.frame_metadata_value.frame_counter)
+        hdr_ts = merged_frame.get_frame_metadata(rs.frame_metadata_value.frame_timestamp)
+        if min_counter > hdr_counter:
+            log.e("min:", min_counter, "hdr:", hdr_counter)
```

Comment:
Will this syntax work in python?
It will not count it as 4 parameters?
I would use
`log.e("min:" + str(min_counter) +  "hdr: " + str(hdr_counter))`
or even better, change the whole condition to
`test.check(min_counter > hdr_counter)`
If you want more detailes about the info in case of failure do:
```
test.info(`min_counter `, min_counter )
test.info(`hdr_counter `, hdr_counter )
test.check(min_counter > hdr_counter)
```

this will print the values only in case of check failure

Reply:
yes I think this syntax should work, because log prints get `*args` as a parameter, the * means it can be any number of parameters as far as I know (like we can do in print)

anyways I like test.info more, I'll switch to that :)